README.md
# cbora12ccl2-cookbook

TODO: Enter the cookbook description here.

## Supported Platforms

TODO: List your supported platforms.

## Attributes

<table>
  <tr>
    <th>Key</th>
    <th>Type</th>
    <th>Description</th>
    <th>Default</th>
  </tr>
  <tr>
    <td><tt>['cbora12ccl2']['bacon']</tt></td>
    <td>Boolean</td>
    <td>whether to include bacon</td>
    <td><tt>true</tt></td>
  </tr>
</table>

## Usage

### cbora12ccl2::default

Include `cbora12ccl2` in your node's `run_list`:

```json
{
  "run_list": [
    "recipe[cbora12ccl2::default]"
  ]
}
```

## License and Authors

Author:: YOUR_NAME (<YOUR_EMAIL>)
# Oracle Client 12c install on Redhat Linux - working version


CHANGELOG.md
# 0.1.0

Initial release of cbora12ccl2


recipes/install_check.rb
#
# Cookbook Name:: cbora12cclt
# Recipe:: install_chefk
#
# Copyright (C) 2016 YOUR_NAME
#
# All rights reserved - Do Not Redistribute
#

::Chef::Resource.send(:include, General::InstallChecks)

oraHome=node[:ora_client12c][:oraHome]

#return only_if { ora_client_already_installed?(oraHome) }

#if ora_client_already_installed (oraHome)

#  Chef::Log.warn("Oracle client exists - NOT CONTINUING" + sqlplusFile)
#  return
#else

#  Chef::Log.info("Oracle client not installed - will install - CONTINUING")
#end

#  Using detailed steps (not using helper)
# Return if already is installed
sqlplusFile=node[:ora_client12c][:oraHome]+'/bin/sqlplus'

if File.exists?(sqlplusFile)
  execute 'Oracle client seems to be already installed.  Check before re-run' do
    command "echo Oracle client seems to be already installed.  Check before re-run > /tmp/oracle_client_installed_check.file"
  end

  return

end

recipes/oracle_client_install_12c.rb
#
# Author: Gowrish.Mallipattana
#
# Oracle client installation recipe
#
#

# Variables set for use by templates and others
oraInventory      =   node[:oracle][:oraInventory]
oraHome           =   node[:ora_client12c][:oraHome]
responseFile      =   node[:ora_client12c][:responseFile] 
softwareFolder    =   node[:ora_client12c][:softwareFolder]
softwareBundle    =   node[:ora_client12c][:softwareBundle]
unzipDir          =   node[:ora_client12c][:unzipDir] 
runInstaller      =   node[:ora_client12c][:unzipDir] + '/client/runInstaller'

# Return if already is installed
if File.exists?(node[:ora_client12c][:oraHome])
  execute 'Oracle client seems to be already installed.  Check before re-run' do
    command "echo Oracle client seems to be already installed.  Check before re-run > /tmp/oracle_client_installed_check.file"
  end
end

return if ::File.exists?(node[:ora_client12c][:oraHome])

# Create the response file from template
template node[:ora_client12c][:responseFile] do
  source 'oracle_client_12c_unix_rsp.erb'
   owner node[:oracle][:userName] 
   group node[:oracle][:installGroup]
    mode '0755'
    variables({
     :installGroup  => node[:oracle][:installGroup],
     :installUser   => node[:ora_client12c][:installUser],
     :oraInventory  => node[:oracle][:oraInventory],
     :oraBase       => node[:oracle][:oraBase],
     :oraHome       => node[:ora_client12c][:oraHome],
     :installType   => node[:ora_client12c][:installType],
     :hostname      => node[:hostname]
  })
end
 
execute 'create unzip directory for oracle software bundle' do
  command "su oracle -c 'mkdir -p #{unzipDir}' "
end

execute 'unzip oracle software bundle' do
  command "su oracle -c 'unzip #{softwareFolder}/#{softwareBundle} -d #{unzipDir}' "
end

execute 'install oracle software bundle' do
  command "su oracle -c '#{runInstaller} -silent -waitforcompletion -noconfig -ignoreSysPrereqs -ignorePrereq -responseFile #{responseFile} > /tmp/out 2>> /tmp/out' "
end

execute 'sleep a while for all filecopy to end' do
  command "sleep 60"
end

execute 'run root shell' do
  user "root"
  command "#{oraHome}/root.sh"
end

execute 'run orainstRoot shell' do
  command "sudo su - root -c '#{oraInventory}/orainstRoot.sh'"
end



recipes/x-hardcoded-oracle_users_groups_dirs_unix.rb
#
# Recipe:: users-and-roles-for-oracle-unix
#
# Copyright (C) 2015 YOUR_NAME
#
# All rights reserved - Do Not Redistribute

# User and roles creation for Oracle installation

# Create groups - begin
# https://docs.chef.io/resource_group.html

# Common install group for all oracle related users
group 'oradba' do
  action :create
  gid '2580'
end

user 'oracle' do
  comment 'Oracle DB and Client software owner'
  uid     '501'
  group   'oradba'
  home    '/home/oracle'
  shell   '/bin/ksh'

  # Note - no password - only do sudo to oracle
  #password 'oracle'
end

# Create users - end

# Create Directories - begin
%w[ /home /u01 /u01/app ].each do |path|
  directory path do
    owner 'root'
    group 'root'
    mode '0755'
    action :create
  end
end

%w[ /u01/app/oraInventory ].each do |path|
  directory path do
    owner 'oracle'
    group 'oradba'
    mode '0775'
    action :create
  end
end

%w[ /home/oracle /u02/app/oracle ].each do |path|
  directory path do
    owner 'oracle'
    group 'oradba'
    mode '0755'
    action :create
  end
end

%w[ /usr/local/tns ].each do |path|
  directory path do
    owner 'oracle'
    group 'oradba'
    mode '0775'
    action :create
  end
end

# Create directories - end


recipes/oracle_users_groups_dirs_unix.rb
#
# Recipe:: users-and-roles-for-oracle-unix
#
# Copyright (C) 2015 YOUR_NAME
#
# All rights reserved - Do Not Redistribute

# User and roles creation for Oracle installation

# Create groups - begin
# https://docs.chef.io/resource_group.html

# Common install group for all oracle related users
group node[:oracle][:installGroup] do
  action :create
  gid    node[:oracle][:groupGid]
end

user node[:oracle][:userName]  do
  comment 'Oracle DB and Client software owner'
  uid     node[:oracle][:userUid] 
  group   node[:oracle][:installGroup]
  home    node[:oracle][:userHome]
  shell   node[:userShell] 

  # Note - no password - only do sudo to oracle
  #password 'oracle'

end

# Create users - end

# Create Directories - begin
%w[ /home /u01 /u01/app /u02 /u02/app ].each do |path|
  directory path do
    owner  'root'
    group  'root'
    mode   '0755'
    action :create
  end
end

dirs = [node[:oracle][:oraInventory]]

dirs.each do |path|
  directory path do
    owner  'oracle'
    group  'oradba'
    mode   '0775'
    action :create
  end
end

#dirs = [node[:oracle][:userHome], node[:oracle][:oraBase], node[:ora_client12c][:softwareFolder]]
dirs = [node[:oracle][:userHome], node[:oracle][:oraBase]]

dirs.each do |path|
  directory path do
    owner  node[:oracle][:userName]
    group  node[:oracle][:installGroup]
    mode   '0755'
    action :create
  end
end

dirs = [node[:oracle][:tnsDir]]
dirs.each do |path|
  directory path do
    owner  node[:oracle][:userName]
    group  node[:oracle][:installGroup]
    mode   '0775'
    action :create
  end
end

# Create directories - end
